features/step_definitions/contracts/contract_steps.rb

```diff
@@ -1,21 +1,22 @@
-Then /^(?:the)? ((?:buyer|provider|account) "[^"]*") should have (?:a|an)(?: (pending|live|suspended))? (account|service|application) contract(?: with the (plan "[^"]*"))?$/ do |account, state, type , plan|
+# frozen_string_literal: true
+
+Then "(the )buyer/provider/{account} should have a/an( {word}) {word} contract with the {plan}" do |account, state, type, plan|
   conditions = {}
   conditions[:state] = state if state.present?
-  conditions[:plans] = {:type => "#{type.camelize}Contract"}
+  conditions[:plans] = { type: "#{type.camelize}Contract" }
   conditions[:plan_id] = plan.id if plan
   conditions[:user_account_id] = account.id
 
   contracts = account.contracts.where(conditions).includes(:plan)
   assert_not_nil contracts
 end
 
-
-When /^the contract of (buyer "[^"]*") with (plan "[^"]*") is approved$/ do |buyer, plan|
+When "the contract of {buyer} with {plan} is approved" do |buyer, plan|
   contract = buyer.contracts.by_plan_id(plan.id).first
   contract.accept!
 end
 
-When /^the provider (accept|suspend|resume)s the buyer's (application|service|service) plan contract$/ do |verb, plan_type|
+When /^the provider (accept|suspend|resume)s the buyer's (account|service|application) plan contract$/ do |verb, plan_type|
```

Comment:
I'm fine to keep regular expression instead of Cucumber expression. Given that you have converted seemingly everything else to cucumber expressions, I'm curious why you didn't use `option1/option2/option3` in a cucumber expression here.